js/store/reducers/error.js

```diff
@@ -22,34 +22,35 @@ export const INITIAL_STATE = {}
  *
  * USAGE: `createErrorSelector(['PROFILE_LOAD', 'PREFERENCES_LOAD'])`
  */
-export const createErrorSelector = (actions: Array<string>) => (
+export const createErrorSelector = (actions: $ReadOnlyArray<string>) => (
   state: GlobalState
-) => {
-  // Returns first error found
-  return (
-    _(actions)
-      .map(action => _.get(state, `error.${action}`))
-      .compact()
-      .first() || ''
-  )
+): ?string => {
+  // Returns first error message found if any
+  return actions
+    .map(action => _.get(state, `error.${action}`))
+    .filter(message => !!message)[0]
 }
 
 // Listen for _REQUEST|_FAILURE actions and set/remove error message.
-const reducer = (state: ErrorState = INITIAL_STATE, action: Action) => {
+const reducer = (state: ErrorState = INITIAL_STATE, action: Action): Object => {
```

Comment:
shouldn't this return a `GlobalState` type?

Reply:
the errorState is just a slice of the GlobalState, i need to add it (and also the loadingState) to the GlobalState like this:

```
export type GlobalState = {
  appState: AppState,
  network: NetworkState,
  navigation: NavigationState,
  user: UserState,
  loading: Object,
  error: Object,
  profile: ProfileState
}
```

Reply:
I mean, the point here is to never ever use `*` or `Object` for types, only in very very very special occasions you can use it...

Reply:
The errorState is managed dynamically by a reducer that add/set/remove variables to the Object with key= [NAME_OF_THE_REQUEST]. The only way to add typing to the errorState Object is something like:

```
type ErrorState = {
PROFILE_LOAD: string, // the string is the error message
MESSAGES_LOAD: string,
...and so on for all the actions request
}
```
but i think we loose flexibility because we need to remember each time we create a new request action to add it to this type. Do you think we have other options?

Reply:
for starters we could have:

```ts
type ErrorState = { 
  [key: string]: string
}
```

and, if you can build a type with all the possible request keys, you could have for instance:

```ts
type RequestType = "PROFILE_LOAD" | "MESSAGE_LOAD" | ...

type ErrorState = { 
  [key: RequestType]: string
}
```

see [here](https://flow.org/en/docs/types/objects/#toc-objects-as-maps), [here](https://flow.org/en/docs/types/unions/#toc-union-type-syntax) and perhaps you may use the [$Keys](https://flow.org/en/docs/types/utilities/) utility type to construct the `RequestType` from some other definition of the requests.